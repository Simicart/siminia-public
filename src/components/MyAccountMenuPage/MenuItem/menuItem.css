.item {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    height: 3.5rem;
    border-bottom: 1px solid rgb(var(--venia-border));
    font-size: 0.875rem;
    font-weight: 300;
    line-height: 0.875rem;
}
