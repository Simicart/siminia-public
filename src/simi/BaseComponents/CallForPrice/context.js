import React, { createContext, useContext } from 'react';
import { useCallForPrice } from './talons/useCallForPrice';

const CallForPriceContext = createContext();
const { Provider } = CallForPriceContext;

const CallForPriceProvider = props => {
    const { children } = props;

    const talonProps = useCallForPrice();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default CallForPriceProvider;

export const useCallForPriceContext = () => useContext(CallForPriceContext);
