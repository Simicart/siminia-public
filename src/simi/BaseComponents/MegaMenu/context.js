import React, { createContext, useContext } from 'react';
import { useMegaMenu } from './talons/useMegaMenu';

const AmMegaMenuContext = createContext();
const { Provider } = AmMegaMenuContext;

const AmMegaMenuProvider = props => {
    const { children } = props;

    const talonProps = useMegaMenu();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default AmMegaMenuProvider;

export const useAmMegaMenuContext = () => useContext(AmMegaMenuContext);
