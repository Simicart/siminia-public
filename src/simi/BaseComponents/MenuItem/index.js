/**
 * Created by codynguyen on 7/10/18.
 */
import React, {Component} from 'react';
import {configColor} from "src/simi/Config";
import PropTypes from 'prop-types';
import defaultClasses from './menuitem.css'
import { mergeClasses } from 'src/classify'

class MenuItem extends Component {
    render() {
        const {title,icon,divider,menuStyle,iconStyle,titleStyle,classes} = this.props;
        const mergedClasses = mergeClasses(classes, defaultClasses)
        return (
            <div className={`${mergedClasses["menu-item-wrap"]} ${divider?mergedClasses['divider']:''}`} style={{borderColor: configColor.menu_line_color}}>
                <div role="presentation" style={menuStyle} onClick={this.props.onClick}>
                    <div className={mergedClasses["menu-content"]}>
                        <div className={mergedClasses["icon-menu"]} style={iconStyle}>
                            {icon}
                        </div>
                        <div className={mergedClasses["menu-title"]} style={titleStyle}>
                            {title}
                        </div>
                    </div>
                </div>
            </div>

        )
    }
}
MenuItem.defaultProps = {
    title : '',
    icon : '',
    divider : true,
    menuStyle : {},
    titleStyle : {},
    iconStyle : {},
    onClick : function () {}
}
MenuItem.propsTypes = {
    classes: PropTypes.object.isRequired,
    title: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.node,
        PropTypes.string,
    ]),
    icon: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.node,
        PropTypes.string,
    ]),
    divider: PropTypes.bool,
    menuStyle : PropTypes.object,
    titleStyle : PropTypes.object,
    iconStyle : PropTypes.object,
    onClick : PropTypes.func
}
export default MenuItem;