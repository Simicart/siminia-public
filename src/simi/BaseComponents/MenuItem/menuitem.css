.menu-item-wrap {
    width: 100%;
    display: flex;
    position: relative;
    box-sizing: border-box;
    text-align: left;
    align-items: center;
    padding-top: 11px;
    padding-bottom: 11px;
    justify-content: flex-start;
    text-decoration: none;
    padding-left: 8px;
    padding-right: 8px;
    cursor: pointer;
}
.menu-item-wrap.divider {
    border-bottom: 1px solid;
}