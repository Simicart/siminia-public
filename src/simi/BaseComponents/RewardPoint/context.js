// import React, { createContext, useContext } from 'react';
// import { useRewardPoint } from './talons/useRewardPoint';

// const BssRewardPointContext = createContext();
// const { Provider } = BssRewardPointContext;

// const BssRewardPointProvider = props => {
//     const { children } = props;

//     const talonProps = useRewardPoint();
//     const { error } = talonProps;

//     if (error) {
//         if (process.env.NODE_ENV !== 'production') {
//             console.error(error);
//         }
//     }

//     const contextValue = {
//         ...talonProps
//     };

//     return <Provider value={contextValue}>{children}</Provider>;
// };

// export default BssRewardPointProvider;

// export const useAmMegaMenuContext = () => useContext(BssRewardPointContext);
