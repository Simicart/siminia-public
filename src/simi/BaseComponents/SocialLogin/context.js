import React, { createContext, useContext } from 'react';
import { useSocialLogin } from './talons/useSocialLogin';
import { useSignInEvent } from './talons/useSignInEvent';

const AmSocialLoginContext = createContext();
const { Provider } = AmSocialLoginContext;

const AmAdvancedReviewsProvider = props => {
  const { children } = props;

  const talonProps = useSocialLogin();
  const { error, needRedirect, redirectUrl, handleErrors } = talonProps;
  const { isBusy } = useSignInEvent({
    needRedirect,
    redirectUrl,
    handleErrors
  });

  if (error) {
    if (process.env.NODE_ENV !== 'production') {
      console.error(error);
    }

    console.log('Check if Amasty modules has been installed!');
  }

  const contextValue = {
    ...talonProps,
    isBusy
  };

  return <Provider value={contextValue}>{children}</Provider>;
};

export default AmAdvancedReviewsProvider;

export const useSocialLoginContext = () => useContext(AmSocialLoginContext);
