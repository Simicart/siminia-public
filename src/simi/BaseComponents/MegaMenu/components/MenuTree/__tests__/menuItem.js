import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import { useAmMegaMenuContext } from '../../../context';
import MenuItem from '../menuItem';

jest.mock('../../MenuTree/menuLink', () => 'MenuLink');
jest.mock('../../MenuTree/SubMenuContent', () => 'SubMenuContent');
jest.mock('../../../context', () => ({
  useAmMegaMenuContext: jest.fn().mockName('useAmMegaMenuContext')
}));

test('renders correct tree', () => {
  useAmMegaMenuContext.mockReturnValue({
    isMobile: true,
    isShowIcons: true,
    config: {}
  });
  const tree = createTestInstance(<MenuItem />);
  expect(tree.toJSON()).toMatchSnapshot();
});
