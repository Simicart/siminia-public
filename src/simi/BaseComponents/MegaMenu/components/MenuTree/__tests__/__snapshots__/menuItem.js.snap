// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct tree 1`] = `
<li
  className="    "
  style={
    Object {
      "--am-mega-menu-submenu-bg-color": undefined,
      "--am-mega-menu-submenu-bg-image": undefined,
      "--am-mega-menu-submenu-text-color": undefined,
      "--am-mega-menu-text-hover-color": undefined,
    }
  }
>
  <MenuLink
    isOpen={false}
    isShowIcons={true}
    level={1}
    openCategoryIds={Set {}}
  />
  <SubMenuContent
    classes={
      Object {
        "main": undefined,
        "root": undefined,
      }
    }
    isMobile={true}
    level={1}
    openCategoryIds={Set {}}
  />
</li>
`;
